src/code/z_camera.c

```diff
@@ -1,3069 +1,6777 @@
+                } else {
+                    anim->unk_16 = eyeAdjustment->yaw;
+                    anim->unk_14 = eyeAdjustment->pitch;
+                }
+                peekAroundPoint.x = anim->collisionClosePoint.x - (atEyeColChk.norm.x + eyeAtColChk.norm.x);
+                peekAroundPoint.y = anim->collisionClosePoint.y - (atEyeColChk.norm.y + eyeAtColChk.norm.y);
+                peekAroundPoint.z = anim->collisionClosePoint.z - (atEyeColChk.norm.z + eyeAtColChk.norm.z);
+                OLib_Vec3fDiffToVecSphGeo(&newEyeAdjustment, at, &peekAroundPoint);
+                newEyeAdjustment.r = eyeAdjustment->r;
+                Camera_Vec3fVecSphGeoAdd(eyeNext, at, &newEyeAdjustment);
+                break;
+            }
 
-        normal3->unk_08 = temp_f2 * NEXTSETTING;
-        // normal3->unk_08 = (f32) ((f32) temp_v0->unk-18 * temp_f2);
+            camera->eye = newEyeColChk.pos;
+            atEyeColChk = newEyeColChk;
 
-        normal3->unk_1C = (65535.0f / 360.0f) * NEXTSETTING + 0.5f;
-        // normal3->unk_1C = (s16) (s32) (((f32) temp_v0->unk-14 * 182.0416717529297f) + 0.5f);
+        case 3:
+        case 6:
+            if (anim->unk_18 != 0) {
+                anim->swingUpdateRateTimer = OREG(52);
+                anim->unk_18 = 0;
+                *eyeNext = *eye;
+            }
 
-        normal3->unk_0C = NEXTSETTING;
-        // normal3->unk_0C = (f32) temp_v0->unk-10;
+            temp_f0 = OLib_Vec3fDist(at, &atEyeColChk.pos);
+            *arg4 = temp_f0 > minDist ? 1.0f : temp_f0 / minDist;
 
-        normal3->unk_10 = NEXTSETTING;
-        // normal3->unk_10 = (f32) temp_v0->unk-C;
+            anim->swingUpdateRate = *arg4 * arg3;
 
-        normal3->unk_14 = NEXTSETTING;
-        // normal3->unk_14 = (f32) temp_v0->unk-8;
+            Camera_Vec3fTranslateByUnitVector(eye, &atEyeColChk.pos, &atEyeColChk.norm, 1.0f);
+            anim->atEyePoly = NULL;
+            if (temp_f0 < OREG(21)) {
+                sp40.yaw = eyeAdjustment->yaw;
+                sp40.pitch = Math_Sins(atEyeColChk.sphNorm.pitch + 0x3FFF) * 16380.0f;
+                sp40.r = (OREG(21) - temp_f0) * PCT(OREG(22));
+                Camera_Vec3fVecSphGeoAdd(eye, eye, &sp40);
+                return;
+            }
+            break;
+        default:
+            if (anim->unk_18 != 0) {
+                anim->swingUpdateRateTimer = OREG(52);
+                *eyeNext = *eye;
+                anim->unk_18 = 0;
+            }
+            anim->swingUpdateRate = arg3;
+            anim->atEyePoly = NULL;
+            eye->x = atEyeColChk.pos.x + atEyeColChk.norm.x;
+            eye->y = atEyeColChk.pos.y + atEyeColChk.norm.y;
+            eye->z = atEyeColChk.pos.z + atEyeColChk.norm.z;
+            break;
+    }
+}
 
-        normal3->unk_18 = NEXTSETTING * 0.01f;
-        // normal3->unk_18 = (f32) ((f32) temp_v0->unk-4 * 0.009999999776482582f);
+s32 Camera_NOP(Camera* camera) {
```

Comment:
I suggest this name, for consistency with Actor_Noop and to better fit the naming scheme.
```suggestion
s32 Camera_Noop(Camera* camera) {
```